specs/waku/waku-rpc-api.md

```diff
@@ -0,0 +1,410 @@
+
+##### Response
+
+- **bool** - `true` or `false` or an error on failure.
+
+#### `waku_subscribe`
+
+Creates and registers a new subscription to receive notifications for inbound waku messages.
+
+##### Parameters
+
+The parameters for this request is an array containing the following fields:
+
+ 1. **string** - The ID of the function call, in case of waku this must contain the value "messages".
+ 2. **object** - The [message filter](#message-filter).
+
+##### Response
+
+- **string** - ID of the subscription or an error on failure.
+
+###### Notifications
+
+Notifications received by the client contain a [message](#message) matching the filter. Below is an example notification:
+
+```json
+{
+  "jsonrpc": "2.0",
+  "method": "shh_subscription",
+  "params": {
+    "subscription": "02c1f5c953804acee3b68eda6c0afe3f1b4e0bec73c7445e10d45da333616412",
+    "result": {
+      "sig": "0x0498ac1951b9078a0549c93c3f6088ec7c790032b17580dc3c0c9e900899a48d89eaa27471e3071d2de6a1f48716ecad8b88ee022f4321a7c29b6ffcbee65624ff",
+      "recipientPublicKey": null,
+      "ttl": 10,
+      "timestamp": 1498577270,
+      "topic": "0xffaadd11",
+      "payload": "0xffffffdddddd1122",
+      "padding": "0x35d017b66b124dd4c67623ed0e3f23ba68e3428aa500f77aceb0dbf4b63f69ccfc7ae11e39671d7c94f1ed170193aa3e327158dffdd7abb888b3a3cc48f718773dc0a9dcf1a3680d00fe17ecd4e8d5db31eb9a3c8e6e329d181ecb6ab29eb7a2d9889b49201d9923e6fd99f03807b730780a58924870f541a8a97c87533b1362646e5f573dc48382ef1e70fa19788613c9d2334df3b613f6e024cd7aadc67f681fda6b7a84efdea40cb907371cd3735f9326d02854",
+      "pow": 0.6714754098360656,
+      "hash": "0x17f8066f0540210cf67ef400a8a55bcb32a494a47f91a0d26611c5c1d66f8c57"
+    }
+  }
+}
+```
+
+#### `waku_unsubscribe`
+
+Cancels and removes an existing subscription.
+
+##### Parameters
+
+ - **string** - The subscription ID.
+
+##### Response
+
+- **bool** - `true` or `false`
+
+#### `waku_newMessageFilter`
+
+Creates a new message filter within the node. This filter can be used to poll for new messages that match the criteria.
+
+##### Parameters
+
+The request must contain a [message filter](#message-filter) as its parameter. 
+
+##### Response
+
+ - **string** - The ID of the filter.
+
+#### `waku_deleteMessageFilter`
+
+Removes a message filter from the node.
+
+##### Parameters
+
+- **string** - ID of the filter created with [`waku_newMessageFilter`](#waku_newMessageFilter).
+
+##### Response
+
+- **bool** - `true` on success or an error on failure.
+
+#### `waku_getFilterMessages`
+
+Retrieves messages that match a filter criteria and were received after the last time this function was called.
+
+##### Parameters
+
+- **string** - ID of the filter created with [`waku_newMessageFilter`](#waku_newMessageFilter).
+
+##### Response
+
+The response contains an array of [messages](#messages) or an error on failure.
+
+#### `waku_post`
+
+The `waku_post` method creates a [waku message](./waku-1.md#messages) and disseminates it to the network.
```

Comment:
Here we should call it envelope. And perhaps link to envelope or say it is send through the messages packet over the network and keep the current link.